editor/scene_tree_dock.cpp

```diff
@@ -1409,9 +1409,102 @@ void SceneTreeDock::fill_path_renames(Node *p_node, Node *p_new_parent, List<Pai
 	_fill_path_renames(base_path, new_base_path, p_node, p_renames);
 }
 
+bool SceneTreeDock::_update_node_path(const NodePath &p_root_path, NodePath &p_node_path, List<Pair<NodePath, NodePath>> *p_renames) {
+	NodePath root_path_new = p_root_path;
+	for (List<Pair<NodePath, NodePath>>::Element *F = p_renames->front(); F; F = F->next()) {
+		if (p_root_path == F->get().first) {
+			root_path_new = F->get().second;
+			break;
+		}
+	}
+
+	// Goes through all paths to check if it's matching.
+	for (List<Pair<NodePath, NodePath>>::Element *F = p_renames->front(); F; F = F->next()) {
+		NodePath rel_path_old = p_root_path.rel_path_to(F->get().first);
+
+		// If old path detected, then it needs to be replaced with the new one.
+		if (p_node_path == rel_path_old) {
+			NodePath rel_path_new = F->get().second;
+
+			// If not empty, get new relative path.
+			if (!rel_path_new.is_empty()) {
+				rel_path_new = root_path_new.rel_path_to(rel_path_new);
+			}
+
+			p_node_path = rel_path_new;
+			return true;
+		}
+
+		// Update the node itself if it has a valid node path and has not been deleted.
+		if (p_root_path == F->get().first && p_node_path != NodePath() && F->get().second != NodePath()) {
+			NodePath abs_path = NodePath(String(root_path_new).plus_file(p_node_path)).simplified();
```

Comment:
Just pushed an extra fix here that fixes errors when moving a Skeleton node with BoneAttachment nodes as its children.

Building the new path now uses `root_path_new` instead of `p_root_path`, otherwise in cases where the root path **and** the child path change, this code would generate a buggy node path relative to the old root.